src/main/java/me/makkuusen/timing/system/Tasks.java

```diff
@@ -46,6 +46,19 @@ private void setParticles(Player player, TrackRegion region) {
         }
         Particle particle;
 
+        if (!region.getSpawnLocation().isWorldLoaded()) {
+            return;
+        }
+
+        if (region.getSpawnLocation().getWorld() != player.getWorld()) {
+            return;
+        }
+
+        if(region.getSpawnLocation().distance(player.getLocation()) > 200) {
+            return;
+        }
```

Comment:
Generalize the override
Make it more like spigot api where you have setCanceled() and isCanceled